lid.cpp

```diff
@@ -30,6 +35,80 @@ void Lid::validate(std::string filePath)
     return;
 }
 
+void Lid::assembleCodeUpdateImage()
+{
+    info("InbandCodeUpdate: assembleCodeUpdateImage\n");
+    utils::subscribeToSystemdSignals(bus);
+    auto method = bus.new_method_call(SYSTEMD_BUSNAME, SYSTEMD_PATH,
+                                      SYSTEMD_INTERFACE, "StartUnit");
+    method.append("assemble-lids.service", "replace");
+    bus.call_noreply(method);
+}
+
+void Lid::unitStateChange(sdbusplus::message_t& msg)
+{
+    uint32_t newStateID{};
+    sdbusplus::message::object_path newStateObjPath;
+    std::string newStateUnit{};
+    std::string newStateResult{};
+
+    // Read the msg and populate each variable
+    msg.read(newStateID, newStateObjPath, newStateUnit, newStateResult);
+
+    auto assembleLidServiceFile = "assemble-lids.service";
+
+    if (newStateUnit == assembleLidServiceFile)
+    {
+        info(
+            "InbandCodeUpdate: Service  - {SERVICEFILE} and Service Return value - {SERVICERESULT}",
+            "SERVICEFILE", newStateUnit, "SERVICERESULT", newStateResult);
+        // unsubscribe to systemd signals
+        utils::unsubscribeFromSystemdSignals(bus);
+
+        if (newStateResult == "done")
+        {
+            info(
+                "InbandCodeUpdate: Assemble Code update image is completed successfully.");
+        }
+        else
+        {
+            // Create the empty version object path
+            error("InbandCodeUpdate: Assemble Code update image Failed");
+            createVersionInterface();
+        }
+    }
+}
+
+void Lid::createVersionInterface()
+{
+    // Creating a Version interface under Updater service to provide the failed
+    // result of assemble code update image task. This result is read by PLDM
+    // service
+    auto versionId = VersionClass::getId(std::to_string(randomGen()));
```

Comment:
nit: use appropriate data type rather than auto?

Reply:
Patrick would reply that 'auto' is preferred since that's one of the main features of the new c++ versions, to allow you to not have to worry about what type the variable is :)